biomage/experiment/download.py

```diff
@@ -0,0 +1,306 @@
+import json
+import os
+from enum import Enum
+from fileinput import filename
+from pathlib import Path
+
+import boto3
+import click
+from botocore.exceptions import ClientError
+
+from ..rds.run import run_rds_command
+from ..utils.constants import (
+    CELLSETS_BUCKET,
+    PROCESSED_FILES_BUCKET,
+    RAW_FILES_BUCKET,
+    SAMPLES_BUCKET,
+)
+from .utils import set_modified_date
+
+output_path = "."
+
+
+s3 = boto3.resource("s3")
+SAMPLES = "samples"
+RAW_FILE = "raw_rds"
+PROCESSED_FILE = "processed_rds"
+CELLSETS = "cellsets"
+
+
+def _download_file(bucket, s3_path, file_path):
+    """
+    Utility to download file
+    """
+    file_path.parent.mkdir(parents=True, exist_ok=True)
+
+    s3_obj = s3.Object(bucket, s3_path)
+    s3_obj.download_file(str(file_path))
+    set_modified_date(file_location=file_path, date=s3_obj.last_modified)
+
+
+def _create_sample_mapping(samples_list, output_path):
+    """
+    Create a mapping of sample names to sample ids.
+    """
+
+    MAPPING_FILE_NAME = "sample_mapping.json"
+
+    samples_file = output_path / MAPPING_FILE_NAME
+
+    sample_mapping = {}
+
+    for sample_name, samples in samples_list.items():
+        sample_mapping[sample_name] = samples[0]["sample_id"]
+
+    samples_file.write_text(json.dumps(sample_mapping))
+
+    print(f"Sample name-id map downloaded to: {str(samples_file)}.\n")
+
+
+def _process_query_output(result_str):
+    """
+    Process query output
+    """
+    json_text = (
+        result_str.replace("+", "")
+        .split("\n", 2)[2]
+        .replace("\n", "")
+        .replace("(1 row)", "")
+        .strip()
+    )
+
+    if not json_text:
+        raise Exception("No data returned from query")
+
+    samples = json.loads(json_text)
+
+    # Create a dictionary of samples
+    result = {}
+
+    for sample in samples:
+
+        if not result.get(sample["sample_name"]):
+            result[sample["sample_name"]] = []
+
+        result[sample["sample_name"]].append(sample)
+
+    return result
+
+
+def _get_samples_v2(experiment_id, input_env):
```

Comment:
I think that you can remove the `v2` to the names of these. Unless in biomage-utils there is a reason why we need to distinguish between v1 and v2?

Reply:
no reason, I'll remove this